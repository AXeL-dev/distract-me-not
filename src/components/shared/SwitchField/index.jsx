import React from 'react';
import { Pane, Switch, Position } from 'evergreen-ui';
import { TooltipLabel, OuterPane } from 'components';

export function SwitchField(props) {
  return (
    <OuterPane display="flex" {...props}>
      <Pane display="flex" alignItems="center" flex={1}>
        <TooltipLabel
          text={props.label}
          size={props.labelSize}
          color={props.labelColor}
          className={props.labelClassName}
          tooltip={props.tooltip}
          tooltipPosition={props.tooltipPosition}
        />
      </Pane>
      <Pane
        display="flex"
        alignItems="center"
        marginRight={props.position === Position.LEFT ? 12 : 0}
      >
        <Switch
          height={props.height || 18}
          checked={props.checked}
          onChange={props.onChange}
        />
      </Pane>
    </OuterPane>
  );
}
