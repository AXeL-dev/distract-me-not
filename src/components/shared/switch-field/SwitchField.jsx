import { Pane, Text, Switch, Position } from 'evergreen-ui';

export default function SwitchField(props) {
  return (
    <Pane
      display="flex"
      padding={props.padding}
      paddingX={props.paddingX || props.padding}
      paddingY={props.paddingY || props.padding}
      paddingTop={props.paddingTop || props.paddingY}
      paddingBottom={props.paddingBottom || props.paddingY}
      paddingLeft={props.paddingLeft || props.paddingX}
      paddingRight={props.paddingRight || props.paddingX}
      margin={props.margin}
      marginX={props.marginX || props.margin}
      marginY={props.marginY || props.margin}
      marginTop={props.marginTop || props.marginY}
      marginBottom={props.marginBottom || props.marginY}
      marginLeft={props.marginLeft || props.marginX}
      marginRight={props.marginRight || props.marginX}
      flexDirection={props.position === Position.LEFT ? 'row-reverse' : 'inherit'}
    >
      <Pane display="flex" alignItems="center" flex={1}>
        <Text>{props.label}</Text>
      </Pane>
      <Pane display="flex" alignItems="center" marginRight={props.position === Position.LEFT ? 12 : 0}>
        <Switch
          height={props.height || 18}
          checked={props.checked}
          onChange={props.onChange}
        />
      </Pane>
    </Pane>
  );
}
