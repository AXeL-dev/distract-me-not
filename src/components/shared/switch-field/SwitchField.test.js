import { render, fireEvent } from "@testing-library/react";
import SwitchField from "./SwitchField";

it('renders correctly', () => {
  const { asFragment } = render(<SwitchField />);
  expect(asFragment()).toMatchSnapshot();
});

it('handles value change', () => {
  const handleChange = jest.fn();
  const { getByRole } = render(<SwitchField checked={false} onChange={handleChange} />);
  const _switch = getByRole('checkbox');
  expect(_switch.checked).toBe(false);
  fireEvent.click(_switch);
  expect(handleChange).toHaveBeenCalledTimes(1);
  //expect(_switch.checked).toBe(true); // will not fire since we are not updating the checked prop state
});
