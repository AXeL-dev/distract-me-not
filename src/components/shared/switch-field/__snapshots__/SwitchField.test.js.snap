// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-nil ub-dspl_flex ub-flx-drct_inherit ub-box-szg_border-box"
  >
    <div
      class="css-nil ub-dspl_flex ub-algn-itms_center ub-flx_1 ub-box-szg_border-box"
    >
      <span
        class="ub-color_425A70 ub-fnt-fam_b77syt ub-fnt-sze_14px ub-f-wght_400 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-box-szg_border-box"
      />
    </div>
    <div
      class="css-nil ub-dspl_flex ub-algn-itms_center ub-mr_0px ub-box-szg_border-box"
    >
      <label
        class="ub-dspl_block ub-w_36px ub-pst_relative ub-box-szg_border-box"
      >
        <input
          class="css-17gzcr6 ub-box-szg_border-box"
          type="checkbox"
        />
        <div
          class="ub-h_18px ub-w_36px ub-box-szg_border-box"
        >
          <div
            class="css-17bhmw6 ub-h_18px ub-w_18px ub-box-szg_border-box"
            data-checked="false"
          >
            <svg
              height="6"
              viewBox="0 0 10 7"
              width="10"
            >
              <path
                d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <div
            class="css-1rprxtl ub-w_36px ub-dspl_flex ub-box-szg_border-box"
            data-checked="false"
          >
            <div
              class="ub-flx_1 ub-pb_2px ub-pl_2px ub-pr_2px ub-pt_2px ub-box-szg_border-box"
            >
              <div
                class="css-13qcn8a ub-w_14px ub-h_14px ub-box-szg_border-box"
              />
            </div>
          </div>
        </div>
      </label>
    </div>
  </div>
</DocumentFragment>
`;
